modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/ScriptManagerBridge.cs

```diff
@@ -427,10 +427,18 @@ static void LookupScriptForClass(Type type)
             // This method may be called before initialization.
             if (NativeFuncs.godotsharp_dotnet_module_is_initialized().ToBool() && Engine.IsEditorHint())
             {
-                foreach (var scriptPath in _pathTypeBiMap.Paths)
+                string[] script_paths = _pathTypeBiMap.Paths.ToArray();
+                if (script_paths.Length > 0)
```

Comment:
Instead of converting to array here, change the type of the [`Paths`](https://github.com/godotengine/godot/blob/1ce6df7087113a61491567f3ac55561d5688e2a8/modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/ScriptManagerBridge.types.cs#L67) property to `IReadOnlyCollection<string>`. That way you can use the `Count` property to check how many script paths there are without allocating.

```suggestion
                IReadOnlyCollection<string> scriptPaths = _pathTypeBiMap.Paths;
                if (scriptPaths.Count > 0)
```